src/Host.cpp

```diff
@@ -628,21 +628,26 @@ void Host::unregisterEventHandler(const QString & name, TScript * pScript )
 
 void Host::raiseEvent( const TEvent & pE )
 {
-    if( pE.mArgumentList.size() < 1 ) return;
-    if( mEventHandlerMap.contains( pE.mArgumentList[0] ) )
```

Comment:
We are passing in a supposedly const reference argument and then the old code was using `operator[ index ]` which is designed to allow __write access__ to what it is used on - this is bad practice IMHO and has crept in from `std` libraries I guess.  In many cases in the Qt Documentation, it directs users to use the (read-only) `at( index )` operator {or the corresponding `value( key )`for key:value type data structures} instead when feasible, which is remarked on as being faster and NOT instantiating default constructed values for intermediate "indexes/keys"...